runtime/parser/parse_model.go

```diff
@@ -237,6 +243,25 @@ func (p *Parser) parseModel(ctx context.Context, node *Node) error {
 	r.ModelSpec.OutputConnector = outputConnector
 	r.ModelSpec.OutputProperties = outputPropsPB
 
+	if r.ModelSpec != nil && len(tmp.Tests) > 0 {
```

Comment:
No need to check `!= nil`, it's guaranteed and if it wasn't, the earlier lines would panic anyway